### PR TITLE
request and response datatypes added to service/client methods

### PR DESCRIPTION
**Pull request by rex-schilasky**
### Description
Server/client registration information still handles type name as combined string of encoding::type_name. This PR introduces new datatype information fields for method request and response type including type name, encoding and descriptor.

These fields will be used in eCAL6 in parallel to the old ones to keep downward compatibility. This PR just introduce the additional fields in the protocol buffer idl files and the registration structs. Serialization tests are fixed.

### CODE REVIEW

File: ecal/core/src/serialization/ecal_serialize_common.cpp [hunk 1]
```diff
@@ -354,6 +354,9 @@ namespace eCAL
         encode_string(pb_method.resp_type, method.resp_type);
         encode_string(pb_method.req_desc, method.req_desc);
         encode_string(pb_method.resp_desc, method.resp_desc);
+        encode_string(pb_method.req_datatype.name, method.req_datatype.name);
+        encode_string(pb_method.req_datatype.encoding, method.req_datatype.encoding);
+        encode_string(pb_method.req_datatype.desc, method.req_datatype.descriptor);
```
Comment (Peguen) [inline ecal/core/src/serialization/ecal_serialize_common.cpp:359]: Response datatype is missing

File: ecal/tests/cpp/serialization_test/src/monitoring_generate.cpp [hunk 2]
```diff
@@ -62,9 +63,7 @@ namespace eCAL
       publisher.tid                  = GenerateString(8);
       publisher.tname                = GenerateString(10);
       publisher.direction            = direction;
-      publisher.tdatatype.name       = GenerateString(8);
-      publisher.tdatatype.encoding   = GenerateString(8);
-      publisher.tdatatype.descriptor = GenerateString(10);
+      publisher.tdatatype            = eCAL::Registration::GenerateDataTypeInformation();
```
Comment (Peguen) [inline ecal/tests/cpp/serialization_test/src/monitoring_generate.cpp:66]: Rename 'publisher' to 'topic'

Comment (rex-schilasky) [reply ecal/tests/cpp/serialization_test/src/monitoring_generate.cpp]: Fixed.

**Review by Peguen — APPROVED** [review]
🌸 great 🌸

**Pull request merged**